include/cppkafka/topic_partition_list.h

```diff
@@ -54,6 +54,16 @@ CPPKAFKA_API TopicPartitionList convert(const TopicPartitionsListPtr& topic_part
 CPPKAFKA_API TopicPartitionList convert(rd_kafka_topic_partition_list_t* topic_partitions);
 CPPKAFKA_API TopicPartitionsListPtr make_handle(rd_kafka_topic_partition_list_t* handle);
 
+// Extracts a partition list subset belonging to the provided topics (case-insensitive)
+CPPKAFKA_API TopicPartitionList make_subset(const TopicPartitionList& partitions,
```

Comment:
This is okay here but I'm not sure about the name, "make_subset" doesn't really explain what the function does. Maybe `find_matches` or something like that? As much as this does return a subset of the input, the important thing is _how_ the subset is created and not the fact that it will return a subset.